src/main/java/bwapi/IdMapperHelper.java

```diff
@@ -0,0 +1,24 @@
+package bwapi;
+
+import java.lang.reflect.Array;
+import java.util.stream.Stream;
+
+final class IdMapperHelper {
+
+  private IdMapperHelper() {}
+
+  static <E extends WithId> E[] toIdTypeArray(Class<E> enumClass) {
```

Comment:
I would really prefer not to use the IdMapperHelper and the WithId interface. At least not in this PR, we might change this in the future.
Just accessing `value` instead of getId should be safe as the variable is declared constant. And the mapping of Id->Enum/Class can be done like in PlayerType.

Reply:
Removed.